corpus/data_augmentation.py

```diff
@@ -245,82 +252,116 @@ def run(self):
             segments_outfile.writelines(segment_file_names)
 
 
-class AddImpulseResponseJob(Job):
+class AudiomentationsJob(Job):
     """
-    Add impulse responses to each recording in the corpus with a given probability to be affected.
-    This is done by convolving the audio with an impulse response to simulate other acoustic conditions.
-    Supports .wav files
-
-    This job is dependent on the audiomentations library.
+    Generic audiomentations job for subclassing. Class not adapted for speed perturbation changes.
     """
 
-    def __init__(
-        self, corpus_file, new_corpus_name, ir_path, augment_prob, time_rqmt=12
-    ):
+    def __init__(self, corpus_file, new_corpus_name):
         """
 
         :param Path corpus_file: Bliss corpus with wav files
         :param str new_corpus_name: name of the new corpus
-        :param str ir_path: path to the directory containing impulse responses
-        :param float augment_prob: probability of each audio to be augmented by impulse response convolution
-        :param int time_rqmt: time requirement for the Sisypus job
+        :param str out_corpus_path: name of the new corpus file
         """
         self.corpus_file = corpus_file
         self.new_corpus_name = new_corpus_name
-        self.ir_path = ir_path
-        self.augment_prob = augment_prob
-
-        self.rqmt = {"time": time_rqmt, "cpu": 2}
 
         self.out_audio_folder = self.output_path("out_audio/", directory=True)
-        self.out_corpus = self.output_path("ir_convoluted.corpus.xml.gz")
+        self.out_corpus = self.output_path("augmented_corpus.xml.gz")
+        self.out_segment_file = self.output_path("augmented.segments")
 
     def tasks(self):
         yield Task("run", rqmt=self.rqmt)
 
-    def run(self):
-        add_impulse_response = audiomentations.AddImpulseResponse(
-            ir_path=self.ir_path, p=self.augment_prob
-        )
-
-        job_id = os.path.basename(self.job_id())
-        orig_corpus = corpus.Corpus()
-        new_corpus = corpus.Corpus()
+    def _run(self, audiomentations_transformation, perturbed_string):
```

Comment:
maybe rename `perturbed_string` to to a more descriptive name e.g.  `perturbed_audio_name_prefix`